modules/EnergyManager/EnergyManager.cpp

```diff
@@ -186,7 +186,7 @@ std::vector<types::energy::EnforcedLimits> EnergyManager::run_optimizer(types::e
     optimized_values.reserve(brokers.size());
 
     for (auto broker : brokers) {
```

Comment:
This should also be `auto&`, otherwise you needlessly copy a `shared_ptr` (note that this copy will also take a lock/mutex in order to internally update the ref count)